packages/@react-stately/data/src/useTreeData.ts

```diff
@@ -319,6 +319,14 @@ export function useTreeData<T extends object>(options: TreeOptions<T>): TreeData
             ...parentNode.children.slice(index)
           ]
         }));
+
+        const collectNodesForMap = (items: TreeNode<T>['children']) => items.forEach(i => {
+          map.set(i.key, i);
```

Comment:
update tree does some map.set/get already

Is there a reason you're fixing it here instead of there? Genuinely curious, I haven't had enough time to really dive into it.

Reply:
should not change map item total count after call `move()`
but map item lost when call `move()` because of `move > updateTree > deleteNode`
so I fix map item after call `move()`, just worked

A better implementation about 'move()' should be considered.

Reply:
After some time today, I think i'd recommend this instead. 
The change should be to line 162
```
    if (newNode == null) {
      deleteNode(node);
    } else {
      addNode(newNode); // this line
    }
```
where addNode is a mirror of deleteNode (deleteNode is responsible for the recursive child removal from the map, so makes sense adding a node should recursively add the children to the map)
along these lines
```
  function addNode(node: TreeNode<T>) {
    map.set(node.key, node);
    for (let child of node.children) {
      addNode(child);
    }
  }
```
then there is no change to the move function and anything else using updateTree doesn't need to worry about adding children
How does that sound?

Reply:
nice. seems to be a good solution.
I don't have time to deal with it. you can keep the unit test to ensure your implementation :)

Reply:
Sure, moving to here https://github.com/adobe/react-spectrum/pull/1465